### PR TITLE
JAMES-2665 Guice bindings for "Deleted Messages vault"

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: server/container/guice/pom.xml [hunk 1]
```diff
@@ -52,6 +52,7 @@
         <module>jpa-smtp-mariadb</module>
         <module>mailbox</module>
         <module>mailbox-plugin-spamassassin</module>
+        <module>mailbox-plugin-deleted-messages-vault-guice</module>
```
Comment (mbaechler) [inline server/container/guice/pom.xml]: ordering issue

File: server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailRepositoryDeletedMessageVault.java [hunk 2]
```diff
@@ -53,6 +59,7 @@ public Configuration(MailRepositoryUrl urlPrefix) {
     private final Configuration configuration;
     private final MailConverter mailConverter;
 
+    @Inject
     public MailRepositoryDeletedMessageVault(MailRepositoryStore mailRepositoryStore, Configuration configuration, MailConverter mailConverter) {
```
Comment (mbaechler) [inline server/mailrepository/deleted-messages-vault-repository/src/main/java/org/apache/james/vault/MailRepositoryDeletedMessageVault.java]: may not be public, then

**Comment by chibenwa** [comment]
:green_apple:  before ci restart

**Comment by chibenwa** [comment]
test this please

A :green_apple:  build confirmation would be nice before a merge

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**